### DISCUSSION TITLE
Packaging error

### DISCUSSION THREAD

**Discussion by LittleTimeWxk** [microsoft/AirSim#4129] [Support Q&A] [answered]
I create a project on Windows, I need to package the project into linux format. But there are errors generated.
UE version: 4.26
AirSim version: 1.5.0

error report:
UATHelper: Pack (Linux):     ld.lld: error: cannot open D:/UnrealProject/MSSimulator/Source/MSSimulator/AirSimSource/AirLib/deps/MavLinkCom/lib/libMavLinkCom.a: no such file or directory
UATHelper: Pack (Linux):     ld.lld: error: cannot open D:/UnrealProject/MSSimulator/Source/MSSimulator/AirSimSource/AirLib/lib/libAirLib.a: no such file or directory
UATHelper: Pack (Linux):     ld.lld: error: cannot open D:/UnrealProject/MSSimulator/Source/MSSimulator/AirSimSource/AirLib/deps/rpclib/lib/librpc.a: no such file or directory
UATHelper: Pack (Linux):     clang++: error: linker command failed with exit code 1 (use -v to see invocation)
UATHelper: Pack (Linux):     Failed to link D:\UnrealProject\MSSimulator\Binaries\Linux\MSSimulator after 10 retries
PackagingResults: Error: cannot open D:/UnrealProject/MSSimulator/Source/MSSimulator/AirSimSource/AirLib/deps/MavLinkCom/lib/libMavLinkCom.a: no such file or directory
PackagingResults: Error: cannot open D:/UnrealProject/MSSimulator/Source/MSSimulator/AirSimSource/AirLib/lib/libAirLib.a: no such file or directory
PackagingResults: Error: cannot open D:/UnrealProject/MSSimulator/Source/MSSimulator/AirSimSource/AirLib/deps/rpclib/lib/librpc.a: no such file or directory
PackagingResults: Error: linker command failed with exit code 1 (use -v to see invocation)

I don’t know how to find these Linux static library files.

**Comment by zimmy87** [comment, marked as answer]
These Linus static libraries should be getting built by build.sh in the repo's root directory. Try running that script before packaging and see if that works.

**Reply by LittleTimeWxk** [reply]
Thanks your reply, I have tried it and get the Linux static libraries.

**Reply by rajat2004** [reply]
Might be helpful to add this info to the docs also for the future

**Reply by ChrisNassif** [reply]
Does this mean that you can only package an airsim project for linux if you are running the linux version of airsim? Because build.sh to my knowledge is only used in the building of airsim on linux is there no way to cross-compile linux on windows?

**Comment by pranay-junare** [comment]
Hi @zimmy87 @LittleTimeWxk,

Thankyou for your extremely fruitful discussion. I ran into the same issue and by build.sh, I am assuming you mean it is the equivalent [build.bat](https://github.com/microsoft/AirSim/blob/main/build.cmd) for Windows machine.

Can you elaborate on how we integrate this build.bat file into the workflow before generating the package from Unreal (before this step from UI : Platform->Linux->Generate Package)

FYI, I have a similar setup, except I am using Unreal 5.1 and a custom Airsim plugin in my Unreal Engine Environment. I am trying to create a cross-compile Linux package of my map from Windows Unreal Engine.
